src/databases/mysql.rs

```diff
@@ -573,6 +592,40 @@ impl Database for Mysql {
         }
     }
 
+    async fn get_torrent_original_info_hashes(&self, canonical: &InfoHash) -> Result<OriginalInfoHashes, database::Error> {
+        let db_info_hashes = query_as::<_, DbTorrentInfoHash>(
+            "SELECT info_hash, canonical_info_hash, original_is_known FROM torrust_torrent_info_hashes WHERE canonical_info_hash = ?",
```

Comment:
add constraint "original is known == TRUE" ?

Reply:
Hi @da2ce7, good point! The way I'm using the function I need the original infohashes plus the canonical infohashes for the unknown originals. That means I'm treating the "unknown" as If they were known and the same as the canonical.

The actual function name should be:

`get_torrent_original_info_hashes_when_is_known_and_the_canonical_when_is_not`

Maybe we can add a parameter to make it explicit:

`async fn get_torrent_original_info_hashes(&self, canonical: &InfoHash, include_unkown: bool) `. What do you think?

The code where the functions is used:

```rust
        let original_info_hashes = self
            .torrent_info_hash_repository
            .get_torrent_original_info_hashes(&canonical_info_hash)
            .await?;

        if !original_info_hashes.is_empty() {
            // Torrent with the same canonical infohash was already uploaded
            debug!("Canonical infohash found: {:?}", canonical_info_hash.to_hex_string());

            if let Some(original_info_hash) = original_info_hashes.find(&original_info_hash) {
                // The exact original infohash was already uploaded
                debug!("Original infohash found: {:?}", original_info_hash.to_hex_string());

                return Err(ServiceError::InfoHashAlreadyExists);
            }

            // A new original infohash is being uploaded with a canonical infohash that already exists.
            debug!("Original infohash not found: {:?}", original_info_hash.to_hex_string());

            // Add the new associated original infohash to the canonical one.
            self.torrent_info_hash_repository
                .add(&original_info_hash, &canonical_info_hash)
                .await?;
            return Err(ServiceError::CanonicalInfoHashAlreadyExists);
        }
```

Reply:
Finally, I've used the name `get_torrent_canonical_info_hash_group`, which is more generic. It returns all the infohashes linked to the same canonical infohash.

Reply:
> I suggest having two functions:
> 
> 1. `get_torrent_original_info_hashes`: that returns a set of original info hashes.
> 2. `get_torrent_info_hashes`, that return the full of info_hashes as tuples `(info_hash, is_original)`.
> 
> What do you think?

We do not need the first one yet. And for the second one, I think I prefer the other name (`get_torrent_canonical_info_hash_group`) because I prefer to introduce a new domain concept instead of a too-generic one. At least we know what we are discussing when we use the new term. And for newcomers, it could be a strange concept since people may expect torrent to have a single info-hash.

Reply:
`get_torrent_canonical_info_hash_group` is a good name :+1: